fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/BaseCompositeAvailabilitySpec.groovy

```diff
@@ -0,0 +1,250 @@
+        availability3.getExpectedEndDate(_ as DataSourceConstraint) >> { expectedEnd_3 }
+
+        availability1.getDataSourceNames() >> ([DataSourceName.of("dsName_1")] as Set<DataSourceName>)
+        availability2.getDataSourceNames() >> ([DataSourceName.of("dsName_2")] as Set<DataSourceName>)
+        availability3.getDataSourceNames() >> ([DataSourceName.of("dsName_3")] as Set<DataSourceName>)
+
+        compositeAvailability = new SimpleCompositeAvailability(
+                Stream.of(
+                        availability1,
+                        availability2,
+                        availability3
+                )
+        )
+    }
+
+    @Unroll
+    def "If #desc availabilities have no start date, the composite has no start date"() {
+        given:
+        expectedStart_1 = Optional.ofNullable((DateTime) testExpectedStart_1)
+        expectedStart_2 = Optional.ofNullable((DateTime) testExpectedStart_2)
+        expectedStart_3 = Optional.ofNullable((DateTime) testExpectedStart_3)
+
+        Optional<DateTime> result_1 = compositeAvailability.getExpectedStartDate(Mock(PhysicalDataSourceConstraint))
+        Optional<DateTime> result_2 = compositeAvailability.getExpectedStartDate(Mock(DataSourceConstraint))
+
+        expect:
+        !result_1.isPresent()
+        !result_2.isPresent()
+
+        where:
+        testExpectedStart_1 |   testExpectedStart_2 |   testExpectedStart_3 |   desc
+        earliestStart       |   middleStart         |   null                |   "one"
+        earliestStart       |   null                |   null                |   "two"
+        null                |   null                |   null                |   "all"
+    }
+
+    @Unroll
+    def "If #desc availabilities have no end date, the composite has no end date"() {
+        given:
+        expectedEnd_1 = Optional.ofNullable((DateTime) testExpectedEnd_1)
+        expectedEnd_2 = Optional.ofNullable((DateTime) testExpectedEnd_2)
+        expectedEnd_3 = Optional.ofNullable((DateTime) testExpectedEnd_3)
+
+        Optional<DateTime> result_1  = compositeAvailability.getExpectedStartDate(Mock(PhysicalDataSourceConstraint))
+        Optional<DateTime> result_2  = compositeAvailability.getExpectedStartDate(Mock(DataSourceConstraint))
+
+        expect:
+        !result_1.isPresent()
+        !result_2.isPresent()
+
+        where:
+        testExpectedEnd_1   |   testExpectedEnd_2   |   testExpectedEnd_3   |   desc
+        earliestEnd         |   middleEnd           |   null                |   "one"
+        earliestEnd         |   null                |   null                |   "two"
+        null                |   null                |   null                |   "all"
+    }
+
+    @Unroll
+    def "earliest start is at position #pos in the availabilities stream"() {
+        given:
+        expectedStart_1 = Optional.ofNullable((DateTime) testExpectedStart_1)
```

Comment:
This test also seems to not do anything with the arguments.

Reply:
same as above